src/lightning_app/components/database/utilities.py

```diff
@@ -0,0 +1,248 @@
+import functools
+import json
+import pathlib
+from typing import Any, Dict, Generic, List, Type, TypeVar
+
+from fastapi import Response, status
+from fastapi.encoders import jsonable_encoder
+from pydantic import BaseModel, parse_obj_as
+from pydantic.main import ModelMetaclass
+
+from lightning_app.utilities.app_helpers import Logger
+from lightning_app.utilities.imports import _is_sqlmodel_available
+
+if _is_sqlmodel_available():
+    from sqlalchemy.inspection import inspect as sqlalchemy_inspect
+    from sqlmodel import JSON, select, Session, SQLModel, TypeDecorator
+
+logger = Logger(__name__)
+engine = None
+
+T = TypeVar("T")
+
+
+# Taken from https://github.com/tiangolo/sqlmodel/issues/63#issuecomment-1081555082
```

Comment:
I think it is great to take inspiration from there for the core logic. However, by copying 1:1 we are inheriting very bad practices. The code should be simplified to be more readable and testable.

Reply:
Yes, unfortunately. This code is slightly magic to me too. I roughly understand how it works and I tested it works in practice.